wrappers/python/nntile/nntile_core.cc

```diff
@@ -307,8 +307,7 @@ void tensor_from_array(const tensor::Tensor<T> &tensor,
 #endif // STARPU_SIMGRID
         tile_local.release();
     }
-    tensor::scatter<T>(tmp, tensor);
-    tmp.unregister();
+    tensor::scatter_async<T>(tmp, tensor);
```

Comment:
On the line 368 please make `gather` also async. It will remove a burden of waiting ALL THE TASKS to finish from the `tensor_to_array`. Only `gather_async` will be awaited by the following `acquire` call.